src/server/src/test/java/io/cassandrareaper/management/http/HttpCassandraManagementProxyTest.java

```diff
@@ -83,4 +109,98 @@ private <T extends Object> T jsonFromResourceFile(String filename, Class<T> claz
     return new ObjectMapper().readValue(
         this.getClass().getResource(filename).openStream(), clazz);
   }
+
+  @Test
+  public void testRepairHandlers() throws Exception {
+    HttpManagementConnectionFactory connectionFactory = Mockito.mock(HttpManagementConnectionFactory.class);
+    DefaultApi mockClient = Mockito.mock(DefaultApi.class);
+    doReturn("repair-123456789").when(mockClient).repair1(any());
+    ScheduledExecutorService executorService = Mockito.mock(ScheduledExecutorService.class);
+    doReturn(ConcurrentUtils.constantFuture(null)).when(executorService).submit(any(Callable.class));
+
+    HttpCassandraManagementProxy httpCassandraManagementProxy = new HttpCassandraManagementProxy(
+        Mockito.mock(MetricRegistry.class), "",
+        Mockito.mock(InetSocketAddress.class), executorService, mockClient);
+    when(connectionFactory.connectAny(any())).thenReturn(httpCassandraManagementProxy);
+
+    RepairStatusHandler repairStatusHandler = Mockito.mock(RepairStatusHandler.class);
+
+    int repairNo = httpCassandraManagementProxy.triggerRepair(BigInteger.ZERO, BigInteger.ONE, "ks",
+        RepairParallelism.PARALLEL,
+        Collections.singleton("table"), true, Collections.emptyList(), repairStatusHandler, Collections.emptyList(), 1);
+
+    assertEquals(123456789, repairNo);
```

Comment:
Issue: can we please test the individual methods here? And also look at what happens under failure conditions?

Reply:
> Issue: can we make some assertions around which functions have been called and how many times?

Done in the next test. It's not what this test would test as it doesn't care if any method is called. This is intended to test that the maps have correct amount of entries (if someone refactors the repair endpoint for example, but forgets to insert correct places, then this test would fail and tell that info).

> Issue: can we please test the individual methods here? And also look at what happens under failure conditions?

This is already calling the individual methods. triggerRepair & removeRepairStatusHandlers are the interesting methods this one calls directly.